# Day 2 In-class Assignment: Order of Magnitude Modeling

### <p style="text-align: right;"> &#9989; **Krizma Nagi** </p>
#### <p style="text-align: right;"> &#9989; Table 2: Mikayla, Delaney, Lexie, Alex, Connor here</p>

## How many mice would it take to fill up this classroom?

### (a.k.a., making order-of-magnitude estimates for fun and profit!)
![Mouse](https://www.jax.org/-/media/jaxweb/images/jax-mice-and-services/mice/web_0009_find-and-order-jax-mice.jpg?h=316&w=467&la=en&hash=D42556D3A616EC2E099378DB9CA5403D0C38C1E4)

## Learning Goals 

Why are we asking you to do this?

One type of modeling is called "order-of-magnitude estimation" or "order-of-magnitude modeling."  A physicist would call this a "Fermi Problem" after Enrico Fermi, a physicist who participated in the Manhattan Project and who was famously good at making estimates of this sort.  More specifically, this sort of modeling makes us think about:

* The quantities that should go into our models, and the reasonable values of those quantities.
* How the different parts of our model might interact with one another.
* What a "good enough" answer would be to a particular problem

## The Problem

In this project, your group is going to calculate an answer one of the following questions:

1. How many grams of caffeine get consumed on MSU's campus each week?  
2. How much does the air above Lake Superior weigh?
3. If somebody left one of the bathroom faucets running at the Breslin Center, how long would it take to fill it up with water?  
4. If all of the toilet paper used per week in the United States were rolled out across the surface of Michigan could you cover the entire state? If so, could you cover it more than once?
5. Which has more mass, the air in this classroom or the brains in this classroom?  
6. On average, how many people are airborne in the world at a given moment?  
7. How many total miles do all Americans drive in their cars in one year, and how does this compare to the circumference of the Earth ($2.5 \times 10^4$ miles), the distance from the Earth to the Sun ($9 \times 10^7$ miles), and size of the Solar System ($10^9$ miles)?
8. After a particularly exciting Spartan football game, the fans rush the field. How many people can fit onto the football field at Spartan Stadium?

---
# Part 1

**As a group**, you will be assigned one of these problems to answer. You will need to come up with:
1. The questions that you need to ask and answer to come up with an estimate for your problem.
2. The equation(s) you will need to solve your problem.
3. The reasonable values that would go into the equation(s).

Once you have finished these steps, you and your group will implement a simple computer program to calculate a range of answers to your problem.

***Use your whiteboard to write out your ideas so your teammates and instructors can see your ideas/progress***

If you have a hybrid team, try a virtual whiteboard app like [Google Jamboard](https://jamboard.google.com/)


## 1.1 Describing Your Model
**In the cell below, write down your assigned question and the parameters in your model.** Also write down the range of possible values for each of the parameters!



What is the volume of the Breslin Center?
At what rate does the water come out of a faucet?

Volume range of Breslin Center = 19050000-38100000 cubic feet
Range of rate of water = 0.267-0.334 cubic feet/minute


## 1.2 Coding Your Model
In the cell below, write a program to calculate and print out an estimate for your model.

> **Important**: When writing your code, **define variables for each value of interest.**  Then use those variables in your equations.  In other words, do not  hard-code the values in your equations (do not put *numbers* in your equations, put *variables* in your equations).  
>
> As an example, if I want to compute and then print the area of a right triangle, which I know to be $A = \frac{1}{2} b h$, I might write my code like so:
>
>> ```
>> base = 10
>> height = 5
>>
>> area = (base * height) / 2
>>
>> print("The area is:", area)
>> ```  
>       
> By doing this, if I decide I need to change the `base` value or the `height` value, I can just change the value where they are defined. This can be especially useful if I have to use the values multiple times or if I want to experiment with changing the values.

In [13]:
# write a program in this cell to calculate and print out an estimate
# of the answer to the question you've chosen to solve!

#high estimate

area = 254000 # floor area of the Breslin Center in square feet
height = 75  # height of the Breslin Center in feet
water_rate = 0.334   # rate of water coming out of faucet in cubic feet per minute
volume = area*height

x= (1/water_rate)*volume

print ("It would take", x, "minutes to fill the Breslin Center.")


It would take 57035928.143712565 minutes to fill the Breslin Center.


## 1.3 Exploring Model Results
**Now, run your model several times**, and each time change the values of the parameters to a different value in the range you think is valid.  In the cell below, write down all of your answers.  What is the total range, and approximately what is the average value?

114071856.3 minutes/Breslin, 71348314.61 minutes/Breslin, 142696629.21 minutes/Breslin Center, 57035928.14 minutes/Breslin Center. This is all the possible times that it could take for the Breslin Center to be filled.

The range goes from 114071856.3 minutes/Breslin to 71348314.61 minutes/Breslin, so the total range is 42723541.69. The average value is 92710085.455 minutes/Breslin. 

---
# Part 2

## 2.1 Pick a Second Model
**Now, picking one of the other questions and going through the process of making another order-of-magnitude estimate.** Each group must choose a *different* question. Once you have decided on which question you are going to try to tackle, try working through the answer on your own, but check in with the members of your group if you run into issues or want to bounce ideas off each other.

## 2.2 Describe the Model
**In the cell below, write down your chosen question and the parameters in your model.** Also write down the range of possible values for each of the parameters!

Question #8
What is the area of the football field?
How much space does a person take?

Area of Spartan Stadium range = 40000-50000 square feet
Area of human being space = 2-6 square feet

## 2.3 Coding the Model

In the cell below, write a program to calculate and print out an estimate for your model.

In [14]:
# write a program in this cell to calculate and print out an estimate
# of the answer to the question you've chosen to solve!

field_area = 50000  #area of spartan stadium in square feet
human_area = 6  #area of human space during covid

x=field_area/human_area

print(x,"people could fill up Spartan Stadium.")


8333.333333333334 people could fill up Spartan Stadium.


## Congratulations, you're done!

Submit this assignment by uploading it to the course Desire2Learn web page.  Go to the "In-class assignments" folder, find the appropriate submission link, and upload it there.

See you next class!

&#169; Copyright 2021,  Michigan State University Board of Trustees